wire-ios-data-model/Tests/Proteus/CryptoboxMigrationManagerTests.swift

```diff
@@ -0,0 +1,179 @@
+
+        // When
+        let result = sut.isMigrationNeeded(accountDirectory: accountDirectory)
+
+        // Then
+        XCTAssertTrue(result)
+    }
+
+    func test_IsMigrationNeeded_FilesExistAndFlagIsOff() {
+        // Given
+        fileManagerMock.cryptoboxFilesExist = true
+        proteusViaCoreCryptoFlag.isOn = false
+
+        // When
+        let result = sut.isMigrationNeeded(accountDirectory: accountDirectory)
+
+        // Then
+        XCTAssertFalse(result)
+    }
+
+    func test_IsMigrationNeeded_FilesDoNotExistAndFlagIsOn() {
+        // Given
+        fileManagerMock.cryptoboxFilesExist = false
+        proteusViaCoreCryptoFlag.isOn = true
+
+        // When
+        let result = sut.isMigrationNeeded(accountDirectory: accountDirectory)
+
+        // Then
+        XCTAssertFalse(result)
+    }
+
+    // MARK: - Perform migration
+
+    func test_itPerformsMigrations() throws {
+        // Given
+        fileManagerMock.cryptoboxFilesExist = true
+        proteusViaCoreCryptoFlag.isOn = true
+
+        mockProteusService.migrateCryptoboxSessionsAt_MockMethod = { _ in }
+
+        // When
+        try sut.performMigration(accountDirectory: accountDirectory, syncContext: syncMOC)
+
+        // Then
+        XCTAssertEqual(fileManagerMock.removeItemInvocations, [cryptoboxDirectory])
+        XCTAssertEqual(mockProteusService.migrateCryptoboxSessionsAt_Invocations.count, 1)
+    }
+
+    func test_itDoesNotPerformMigration_CoreCryptoError() {
+        // Given
+        fileManagerMock.cryptoboxFilesExist = true
+        proteusViaCoreCryptoFlag.isOn = true
+
+        mockProteusService.migrateCryptoboxSessionsAt_MockError = CryptoboxMigrationManager.Failure.failedToMigrateData
+
+        // When
+        XCTAssertThrowsError(try sut.performMigration(accountDirectory: accountDirectory, syncContext: syncMOC)) { error in
+            XCTAssertEqual(error as? CryptoboxMigrationManager.Failure, CryptoboxMigrationManager.Failure.failedToMigrateData)
+        }
+
+        // Then
+        XCTAssertTrue(fileManagerMock.removeItemInvocations.isEmpty)
+    }
+
+    // MARK: - Complete migration
+
+    func test_itCompletesMigration_FlagIsOn() throws {
+        // Given
+        proteusViaCoreCryptoFlag.isOn = true
+        mockProteusService.completeInitialization_MockMethod = {}
+
+        // When
+        try sut.completeMigration(syncContext: syncMOC)
+
+        // Then
+        XCTAssertEqual(mockProteusService.completeInitialization_Invocations.count, 1)
+    }
+
+    func test_itCompletesMigration_FlagIsOff() throws {
+        // Given
+        proteusViaCoreCryptoFlag.isOn = false
+        mockProteusService.completeInitialization_MockMethod = {}
+
+        // When
+        try sut.completeMigration(syncContext: syncMOC)
+
+        // Then
+        XCTAssertEqual(mockProteusService.completeInitialization_Invocations.count, 0)
+    }
+
+}
+
+class FileManagerMock: FileManagerInterface {
```

Comment:
make private or move to own file

+ is there a reason for not using sourcery?

Reply:
I think I wasn't initially in data model so did it manually, but I could use it now

Reply:
auto mocked this and moved to a new file